aronnax/core.py

```diff
@@ -158,6 +171,18 @@ def v_point_variable_3d(grid, func):
             v_variable_3d[i,:,:] = f(X, Y)
     return v_variable_3d
 
+def time_series_variable(nTimeSteps, dt, func):
+    '''Input generator for a time series variable. If passed a function, then that function can depend on the number of timesteps, `nTimeSteps`, and the timestep, `dt`.'''
+
+    ts_variable = np.zeros((nTimeSteps))
+
+    for i, f in enumerate(func):
```

Comment:
This loop looks odd.  It iterates down the `func` list, checks whether the list itself is an int, long, or float, which it can't be if it's a list, and then sets ts_variable to the result of calling each list element in turn, thereby keeping only the last one.  What is this trying to do?  Does it actually work?  This would be an excellent bit of code to add a unit test for.